src/main/java/com/matera/cadastrocentral/telephone/TelephoneService.java

```diff
@@ -42,26 +44,9 @@ public Telephone insertTelephone(UUID clientId, TelephoneDTO telephone) throws T
         }
     }
 
-    public void deleteTelephone(UUID telephoneId) {
-        Optional<Telephone> telephoneOptional = telephoneRepository.findById(telephoneId);
-        if (telephoneOptional.isPresent()) {
-            telephoneRepository.deleteById(telephoneId);
-        } else {
-            throw new TelephoneNotFound();
-        }
-    }
-
-    public Optional<Telephone> findByClientIdAndTelephoneId(UUID clientId, UUID telephoneId) {
-        List<Telephone> clientList = findAllTelephonesByClientId(clientId);
-        Stream<Telephone> telephoneStream = clientList.stream().filter(telephone -> telephone.getTelephoneId().equals(telephoneId));
-        Optional<Telephone> telephoneSpecific = telephoneStream.findFirst();
-        if (telephoneSpecific.isPresent()) {
-            return telephoneSpecific;
-        } else if (clientList.isEmpty()) {
-            throw new ClientNotFound();
-        } else {
-            throw new TelephoneNotFound();
-        }
+    public void deleteTelephone(UUID clientId, UUID telephoneId) {
+        telephoneRepository.delete(telephoneRepository.findByTelephoneIdAndClientId(clientId, telephoneId)
+                .orElseThrow(() -> new TelephoneNotFound()));
     }
 
     public Telephone alterTelephoneByTelephoneId(TelephoneDTO telephone) {
```

Comment:
You will need to change this method here too. It will need to get the `optionalTelephone` by the `findByTelephoneIdAndClientId`

Reply:
Doesn't the .orElseThrow() method allready deal with that?

Reply:
you can do the orelsethrow here too like

```
Telephone telephone = elephoneRepository.findByTelephoneIdAndClientId(clientId, telephoneId)
                .orElseThrow(() -> new TelephoneNotFound())
// if it is present contuinue with the update
```